### PR TITLE
Fix JDT synchronization issues

### PR DESCRIPTION
**Pull request by Feuermagier**
### Checklist
- [x] I tested *all* changes and their related features locally or with a test instance of Artemis (as far as I can test them).
- [x] I documented the Java code using JavaDoc style.
- [x] I documented the changes in the Documentation (/docs).

### Motivation and Context
See #270. The race conditision does still occure even after my last PR. This PR is a draft as more testing (ideally by as many people as possible) is required.

### Description
Trigger an incremental build of the project before opening packages or files. This should force the JDT to completely initialize itself.

### Steps for Testing
Start many assessments and watch for unexpected compilation failures because of wrong package declarations. It is also a failure if no packages are opened automatically, as this indicates that our plugin code ran before the JDT has identified any packages.

### CODE REVIEW

**Comment by Feuermagier** [comment]
> Maybe the problem is that it conflicts with the UpdateMavenProject Job. Maybe we should open the packages after the job is finished.

You are probably right. I've added a callback system for the build so that the UI can get notified when the build is completed.

File: bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/eclipse/common/core/artemis/WorkspaceUtil.java [hunk 1]
```diff
@@ -56,7 +62,15 @@ public static final void createEclipseProject(final String projectName) throws C
...
 
-		new UpdateMavenProjectJob(List.of(project)).schedule();
+		var job = new UpdateMavenProjectJob(List.of(project));
+		job.addJobChangeListener(new JobChangeAdapter() {
+			@Override
+			public void done(IJobChangeEvent e) {
+				buildCallbacks.forEach(c -> c.onPhaseCompleted(project));
+			}
+		});
+		;
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.common.core/src/edu/kit/kastel/eclipse/common/core/artemis/WorkspaceUtil.java:72]:
```suggestion
```

File: bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java [hunk 2]
```diff
@@ -475,58 +472,45 @@ private void resetCombos() {
...
-						} else {
-							LOG.warn("No main class found");
-						}
-					}
-				} catch (JavaModelException e) {
-					LOG.error("JDT failure", e);
+					// ... and focus it in the package explorer
+					Display.getDefault().asyncExec(() -> {
+						explorer.ifPresent(e -> e.selectReveal(new StructuredSelection(mainType.get().getResource())));
+					});
```
Comment (dfuchss) [inline bundles/edu.kit.kastel.sdq.eclipse.grading.view/src/edu/kit/kastel/eclipse/grading/view/assessment/ArtemisGradingView.java:506]: No block needed

**Review by dfuchss — COMMENTED** [review]
Some minor comments :)

**Pull request closed without merging**